tests/integration/olp-cpp-sdk-dataservice-read/VolatileLayerClientCacheTest.cpp

```diff
@@ -135,13 +122,18 @@ TEST_P(VolatileLayerClientCacheTest, GetVolatilePartitionsExpiry) {
   ASSERT_EQ(1u, partitions_response.GetResult().GetPartitions().size());
 
   // wait for the layer to expire in cache
-  std::this_thread::sleep_for(std::chrono::seconds(2));
-  request.WithFetchOption(olp::dataservice::read::OnlineIfNotFound);
-  future = catalog_client->GetPartitions(request);
-  partitions_response = future.GetFuture().get();
-
-  ASSERT_TRUE(partitions_response.IsSuccessful())
-      << ApiErrorToString(partitions_response.GetError());
+  bool expired = false;
+  while(true) {
+    request.WithFetchOption(olp::dataservice::read::CacheOnly);
+    future = catalog_client->GetPartitions(request);
+    partitions_response = future.GetFuture().get();
+    if (!partitions_response.IsSuccessful()) {
+      expired = true;
+      break;
+    }
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+  }
+  ASSERT_TRUE(expired);
   ASSERT_EQ(0u, partitions_response.GetResult().GetPartitions().size());
```

Comment:
I guess you can safely remove this as this is not part of the test itself. You already test in line 130 that the requests was not successful so I would expect that the result is empty.